src/main/java/com/exasol/udfstartuptimeimprover/UdfStartUpTimeImprover.java

```diff
@@ -0,0 +1,171 @@
+
+    private UdfStartUpTimeImprover() {
+        // static class
+    }
+
+    /**
+     * Entrypoint of the startup time improver UDF.
+     *
+     * 
+     * @param exaMetadata exasol metadata
+     * @param exaIterator iterator
+     * 
+     * @throws Exception if something goes wrong
+     */
+    @SuppressWarnings({ "java:S112", "java:S1130" }) // Exception is too generic and not thrown. This signature is
+    // however given by the UDF framework
+    public static String run(final ExaMetadata exaMetadata, final ExaIterator exaIterator) throws Exception {
+        return runInternal(exaMetadata, exaIterator);
+    }
+
+    private static String runInternal(final ExaMetadata exaMetadata, final ExaIterator exaIterator) {
+        final String udfDefinitionString = readStringParameter(exaIterator, PARAMETER_UDF_DEFINITION);
+        final UdfDefinition udfDefinition = new UdfDefinitionParser().parseUdfDefinition(udfDefinitionString);
+        final Path pathToClassList = Path.of(readStringParameter(exaIterator, PARAMETER_CLASSES));
+        final String connectionName = readStringParameter(exaIterator, PARAMETER_CONNECTION);
+        final String bfsService = readStringParameter(exaIterator, PARAMETER_BUCKET_FS_SERVICE);
+        final String bfsBucket = readStringParameter(exaIterator, PARAMETER_BUCKET_FS_BUCKET);
+        final String pathForDump = readStringParameter(exaIterator, PARAMETER_PATH_FOR_DUMP);
+        final int bfsPort = readIntParameter(exaIterator, PARAMETER_BUCKET_FS_PORT);
+        final ExaConnectionInformation bucketFsConnection = getConnection(exaMetadata, connectionName);
+        assertClassListExists(pathToClassList);
+        runClassDumpGeneration(udfDefinition.getJars(), pathToClassList);
+        final UnsynchronizedBucket bucket = WriteEnabledBucket.builder().useTls(false).raiseTlsErrors(false)
+                .ipAddress("127.0.0.1").port(bfsPort).serviceName(bfsService).name(bfsBucket).readPassword("")
+                .writePassword(bucketFsConnection.getPassword()).build();
+        uploadDump(bucket, pathForDump);
+        return rewriteUdfDefinition(udfDefinition, bfsService, bfsBucket, pathForDump);
+    }
+
+    private static String rewriteUdfDefinition(final UdfDefinition udfDefinition, final String bfsService,
+            final String bfsBucket, final String pathForDump) {
+        final Path pathToDumpForInUdf = Path.of("/buckets/").resolve(bfsService).resolve(bfsBucket)
+                .resolve(pathForDump);
+        final ArrayList<String> newJvmOptions = new ArrayList<>(udfDefinition.getJvmOptions());
+        newJvmOptions.add("-XX:SharedArchiveFile=" + pathToDumpForInUdf);
+        final String udfDefWithJvmOption = udfDefinition.withJvmOptions(newJvmOptions).toString();
+        if (!udfDefWithJvmOption.startsWith("CREATE OR REPLACE")) {
+            return "CREATE OR REPLACE" + udfDefWithJvmOption.substring("CREATE".length());
```

Comment:
Is `udfDefWithJvmOption` always in upper case? If nessesary trim it and convert it to upper case before.

Reply:
:+1: good catch